### PR TITLE
Httpclient smithy

### PR DESCRIPTION
**Pull request by stevelr**
Addresses issue #116 port http-client widl interface to smithy

### CODE REVIEW

**Comment by stevelr** [comment]
sorry about the messy branch history. This branch is based on main and is independent of the builtin/numbergen PR

File: http-client/rust/src/lib.rs [hunk 1]
```diff
@@ -1,58 +1,4 @@
...
-//! }
-//!
-//! # fn request(method: String, url: String, headers: std::collections::HashMap<String,String>, body: Vec<u8>) -> HandlerResult<httpclient::Response> {
-//! #   Ok(httpclient::Response {
-//! #     status: "OK".to_string(),
-//! #     status_code: 200,
-//! #     ..Default::default()
-//! #   })
-//! # }
+//! org.wasmcloud.provider.httpclient
```
Comment (autodidaddict) [inline http-client/rust/src/lib.rs:1]: This crate is going to be published on crates[dot]io, so it should have a full set of module-level documentation just like the old module, which should include an example of what it looks like to create an actor that uses this module.

**Review by autodidaddict — CHANGES_REQUESTED** [review]

**Comment by autodidaddict** [comment]
You'll also need to fix the DCO status

**Comment by brooksmtownsend** [comment]
@stevelr where are we with this PR? Is it still being worked on?

**Comment by stevelr** [comment]
It's still open but paused because it's slightly  lower priority than getting httpserver, keyvalue, and numbergen working with unit tests, documentation, and sample actor code.

**Comment by stevelr** [comment]
converted to draft pr

**Pull request closed without merging**